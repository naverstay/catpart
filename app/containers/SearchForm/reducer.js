/*
 * searchReducer
 *
 * The reducer takes care of our data. Using actions, we can
 * update our application state. To add a new action,
 * add it to the switch statement in the reducer function
 *
 */

import produce from "immer";
import { CHANGE_ARTNUMBER } from "./constants";

// The initial state of the App
export const initialState = {
  artNumber: ""
};

/* eslint-disable default-case, no-param-reassign */
const searchReducer = (state = initialState, action) =>
  produce(state, draft => {
    switch (action.type) {
      case CHANGE_ARTNUMBER:
        // Delete prefixed '@' from the github artNumber
        draft.artNumber = action.artNumber.replace(/@/gi, "");
        break;
    }
  });

export default searchReducer;
