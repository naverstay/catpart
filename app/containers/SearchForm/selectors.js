/**
 * SearchForm selectors
 */

import { createSelector } from 'reselect';
import { initialState } from './reducer';

const selectHome = state => state.home || initialState;

const makeSelectArtNumber = () =>
  createSelector(
    selectHome,
    homeState => homeState.artNumber,
  );

export { selectHome, makeSelectArtNumber };
