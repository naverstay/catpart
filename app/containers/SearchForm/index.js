/*
 * SearchForm
 *
 */

import React, {useEffect, memo, useState} from 'react';
import PropTypes from 'prop-types';
import {Helmet} from 'react-helmet';
//import {FileDrop} from 'react-file-drop';
import {FormattedMessage} from 'react-intl';
import {connect} from 'react-redux';
import {compose} from 'redux';
import {createStructuredSelector} from 'reselect';
import {readFile} from '../../utils/fileReader';

import {useInjectReducer} from 'utils/injectReducer';
import {useInjectSaga} from 'utils/injectSaga';
import {
  makeSelectRepos,
  makeSelectLoading,
  makeSelectError
} from 'containers/App/selectors';
import H2 from 'components/H2';
import ReposList from 'components/ReposList';
import AtPrefix from './AtPrefix';
import CenteredSection from './CenteredSection';
import Form from './Form';
import Input from './Input';
import Section from './Section';
import messages from './messages';
import {loadRepos} from '../App/actions';
import {changeUsername} from './actions';
import {makeSelectUsername} from './selectors';
import {setInputFilter} from '../../utils/inputFilter';
import reducer from './reducer';
import saga from './saga';

const key = 'home';

export function SearchForm({
                             dndFile,
                             notificationFunc,
                             onSubmitForm,
                             username,
                             loading,
                             error,
                             repos,
                             onChangeUsername
                           }) {
  useInjectReducer({key, reducer});
  useInjectSaga({key, saga});

  const [centeredForm, setCenteredForm] = useState(true);
  const [formBusy, setFormBusy] = useState(true);
  const [formDrag, setFormDrag] = useState(false);

  let formRef = React.createRef();
  let formArtNumber = React.createRef();
  let formQuantity = React.createRef();
  let formFile = React.createRef();

  let onSubmitSearchForm = (values) => {


  };

  useEffect(() => {
    setInputFilter(formQuantity.current, function (value) {
      return /^\d*\.?\d*$/.test(value); // Allow digits and '.' only, using a RegExp
    });
  }, [])

  let onFinish = values => {
    setFormBusy(true);

    setTimeout(() => {
      setFormBusy(false);
    }, 200)
  };

  let onReset = () => {
    formRef.current.resetFields();
  };

  let onFill = () => {
    formRef.current.setFieldsValue({
      note: 'Hello world!',
      gender: 'male'
    });
  };

  const reposListProps = {
    loading,
    error,
    repos
  };

  return (
    <div className={'form-search'}>
      <form ref={formRef} className={'form-content'} onSubmit={onSubmitForm}>
        <div className="form-search__title">
          Поиск электронных <br/> компонентов
        </div>

        <div className="row">
          <div className="form-cell column sm-col-12 md-col-5 lg-col-4 xl-col-3">
            <label className={'form-label'} htmlFor="art-number">Номер компонента</label>
            <div className="form-control">
              <input ref={formArtNumber} id={'art-number'} type="text" className="input __lg"/>
            </div>
            <div className="form-tip">
              Например, <span className={'form-tip__example'} onClick={() => {
              formArtNumber.current.value = 'MAX34'
            }}>MAX34</span>
            </div>
          </div>

          <div className="form-cell column sm-col-12 md-col-3 lg-col-2">
            <label className={'form-label'} htmlFor="quantity">Количество</label>
            <div className="form-control">
              <input ref={formQuantity} id={'quantity'} type="text" className="input __lg"/>
            </div>
            <div className="form-tip">
                <span className={'form-tip__example'} onClick={() => {
                  formQuantity.current.value = '100'
                }}>100</span>
              <span className={'form-tip__example'} onClick={() => {
                formQuantity.current.value = '250'
              }}>250</span>
              <span className={'form-tip__example'} onClick={() => {
                formQuantity.current.value = '500'
              }}>500</span>
            </div>
          </div>

          <div className="form-cell form-cell__search column sm-col-12 md-col-4 lg-col-2">
            <span className={'form-label'}>&nbsp;</span>
            <div className="form-control">
              <button className="btn __blue __lg __w-100p">
                <span>Искать</span>
              </button>
            </div>
          </div>

          <div className="form-cell column form-cell__or sm-col-12 md-col-4 lg-col-2">
            <span className={'form-label'}>&nbsp;</span>
            <div data-or={'или'} className="form-control">
              <label className="btn __lg __gray-dash __w-100p">
                <span>Загрузить BOM</span>
                <input className={'hide'} ref={formFile} id="file" type="file" onChange={() => {
                  readFile(formFile.current.files[0], (ret) => {
                    console.log('readFile', ret);

                    if (ret.success) {

                    } else {

                    }

                  })
                }}/>
              </label>
            </div>
            <div className="form-tip">Выберите или перетащите сюда файл</div>
          </div>
        </div>

        <label style={{position: 'absolute'}} htmlFor="username">
          <Input
            id="username"
            type="text"
            placeholder="mxstbr"
            value={username}
            onChange={onChangeUsername}
          />
        </label>
      </form>
    </div>
  );
}

SearchForm.propTypes = {
  dndFile: PropTypes.string,
  loading: PropTypes.bool,
  error: PropTypes.oneOfType([PropTypes.object, PropTypes.bool]),
  repos: PropTypes.oneOfType([PropTypes.array, PropTypes.bool]),
  notificationFunc: PropTypes.func,
  onSubmitForm: PropTypes.func,
  username: PropTypes.string,
  onChangeUsername: PropTypes.func
};

const mapStateToProps = createStructuredSelector({
  repos: makeSelectRepos(),
  username: makeSelectUsername(),
  loading: makeSelectLoading(),
  error: makeSelectError()
});

export function mapDispatchToProps(dispatch) {
  return {
    onChangeUsername: evt => dispatch(changeUsername(evt.target.value))
    //onSubmitForm: evt => {
    //  console.log('onSubmitForm');
    //  if (evt !== undefined && evt.preventDefault) evt.preventDefault();
    //  dispatch(loadRepos());
    //}
  };
}

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps
);

export default compose(
  withConnect,
  memo
)(SearchForm);
