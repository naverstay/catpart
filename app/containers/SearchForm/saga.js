/**
 * Gets the repositories of the user from Github
 */

import { call, put, select, takeLatest } from 'redux-saga/effects';
import { LOAD_REPOS } from 'containers/App/constants';
import { reposLoaded, repoLoadingError } from 'containers/App/actions';

import request from 'utils/request';
import { makeSelectArtNumber } from 'containers/SearchForm/selectors';

/**
 * Github repos request/response handler
 */
export function* getRepos() {
  // Select username from store
  const artNumber = yield select(makeSelectArtNumber());
  //const requestURL = `https://api.github.com/users/${username}/repos?type=all&sort=updated`;
  const requestURL = `http://dev.catpart.ru/api/search`;

  let options = {
    q: artNumber,
    c: 1,
  };

  try {
    // Call our request helper (see 'utils/request')
    const repos = yield call(request, requestURL, options);
    yield put(reposLoaded(repos, artNumber));
  } catch (err) {
    yield put(repoLoadingError(err));
  }
}

/**
 * Root saga manages watcher lifecycle
 */
export default function* githubData() {
  // Watches for LOAD_REPOS actions and calls getRepos when one comes in.
  // By using `takeLatest` only the result of the latest API call is applied.
  // It returns task descriptor (just like fork) so we can continue execution
  // It will be cancelled automatically on component unmount
  yield takeLatest(LOAD_REPOS, getRepos);
}
