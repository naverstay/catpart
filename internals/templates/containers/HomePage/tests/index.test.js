import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';

import SearchForm from '../index';

describe('<SearchForm />', () => {
  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <SearchForm />
      </IntlProvider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
