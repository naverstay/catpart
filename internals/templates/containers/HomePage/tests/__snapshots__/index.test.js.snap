// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchForm /> should render and match the snapshot 1`] = `
<h1>
  <span>
    This is the SearchForm container!
  </span>
</h1>
`;
